chat/public/js/src/chat.js

```diff
@@ -265,6 +265,39 @@ function ChatXBlock(runtime, element, init_data) {
     var last_sound_played;
 
     /**
+     * localStorageKey: returns a key under which state for this block instance
+     * is stored in localStorage.
+     */
+    var localStorageKey = function() {
+        return 'chat-xblock-' + init_data["block_id"];
+    };
+
+
+    /**
+     * getStateFromLocalStorage: returns state saved in local storage, or null if it does not exist.
+     */
+    var getStateFromLocalStorage = function() {
+        var state = null;
+        try {
+            state = JSON.parse(localStorage.getItem(localStorageKey()));
+        } catch (e) {
+            // ignore
```

Comment:
`localStorage` will throw a security exception in some cases - for example if the browser is configured to not allow using local storage, and I think some browsers will also throw an exception when in private mode.
I will add a comment to explain that and take `JSON.parse` out of the `try-catch` block.